code/modules/modular_computers/file_system/programs/generic/camera.dm

```diff
@@ -1,67 +1,54 @@
-// Returns which access is relevant to passed network. Used by the program.
-/proc/get_camera_access(var/network)
-	if(!network)
-		return 0
-	. = global.using_map.get_network_access(network)
-	if(.)
-		return
-
-	switch(network)
-		if(NETWORK_ENGINEERING, NETWORK_ALARM_ATMOS, NETWORK_ALARM_CAMERA, NETWORK_ALARM_FIRE, NETWORK_ALARM_POWER)
-			return access_engine
-		if(NETWORK_CRESCENT, NETWORK_ERT)
-			return access_cent_specops
-		if(NETWORK_MEDICAL)
-			return access_medical
-		if(NETWORK_MINE)
-			return access_mailsorting // Cargo office - all cargo staff should have access here.
-		if(NETWORK_RESEARCH)
-			return access_research
-		if(NETWORK_THUNDER)
-			return 0
-
-	return access_security // Default for all other networks
-
 /datum/computer_file/program/camera_monitor
 	filename = "cammon"
 	filedesc = "Camera Monitoring"
 	nanomodule_path = /datum/nano_module/program/camera_monitor
 	program_icon_state = "cameras"
 	program_key_state = "generic_key"
 	program_menu_icon = "search"
-	extended_desc = "This program allows remote access to the camera system. Some camera networks may have additional access requirements."
+	extended_desc = "This program allows remote access to the camera system. Some cameras may have additional access requirements."
 	size = 12
 	available_on_network = 1
+	requires_network = 1
```

Comment:
Do you have to set this? Its behaviour is pretty gross, just killing program whenever network connection is lost, mostly left on programs that can't handle gracefully not getting any data or network.

Reply:
Resolved. The program itself will still require a network connection to connect to any cameras, even for television cameras, which makes sense to me but this can be changed if desired.